pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -39,6 +38,91 @@ func newReconciler(scope *machineScope) *Reconciler {
 	}
 }
 
+var (
+	supportedGpuTypes = map[string]string{
+		"nvidia-tesla-k80":  "NVIDIA_K80_GPUS",
+		"nvidia-tesla-p100": "NVIDIA_P100_GPUS",
+		"nvidia-tesla-v100": "NVIDIA_V100_GPUS",
+		"nvidia-tesla-a100": "NVIDIA_A100_GPUS",
+		"nvidia-tesla-p4":   "NVIDIA_P4_GPUS",
+		"nvidia-tesla-t4":   "NVIDIA_T4_GPUS",
+	}
+)
+
+func containsString(sli []string, str string) bool {
+	for _, elem := range sli {
+		if elem == str {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *Reconciler) checkQuota(machineFamily int64) error {
```

Comment:
Maybe this would be a more obvious as a name for this variable as to what this number really is?
```suggestion
func (r *Reconciler) checkQuota(machineTypeAcceleratorCount int64) error {
```